src/dht/findprovs.js

```diff
@@ -23,6 +23,12 @@ module.exports = (send) => {
     }
 
     const handleResult = (res, callback) => {
+      // callback with an empty array if no providers are found
+      if (res === null) {
```

Comment:
According to the issue, `res` is `undefined` so this change won't resolve the issue. Suggestion below (which will catch both `null` _and_ `undefined`):

```suggestion
      if (res == null) {
```

A small unit test to assert this, that mocks out `send` would have caught this!

Reply:
or instead of double equals, we can do `if (!res) { ... }`